corehq/apps/users/views/utils.py

```diff
@@ -115,58 +116,87 @@ def log_commcare_user_locations_changes(request, user, old_location_id, old_assi
         )
 
 
-def _get_location_ids_with_single_user(domain, location_ids, user_id):
-    # Remove the user's ID, as we want to see which other CommCare users share any of the given location_ids
-    user_query = (
+def _flatten_list(list_2d):
+    flat_list = []
+    # Iterate through the outer list
+    for element in list_2d:
+        if type(element) is list:
+            # If the element is of type list, iterate through the sublist
+            for item in element:
+                flat_list.append(item)
+        else:
+            flat_list.append(element)
+    return flat_list
+
+
+def _get_location_ids_with_other_users(domain, location_ids, user_id):
+    """
+    Gets all the location ids where a `CommCareUser` is assigned to one of the `location_ids`.
+    This list excludes the given `user_id`.
+    """
+    other_user_loc_ids = (
         UserES()
         .domain(domain)
         .mobile_users()
+        .NOT(filters.doc_id(user_id))
         .location(location_ids)
+        .values_list('assigned_location_ids', flat=True)
     )
-    user_query.filter(filters.NOT(filters.doc_id(user_id)))
-    other_commcare_user_ids = user_query.get_ids()
+    # If more than one user shares a location, it will give a list of lists
+    # So need to make sure to flatten these
+    return set(_flatten_list(other_user_loc_ids))
+
 
-    case_query = (
+def _get_location_case_counts(domain, location_ids):
+    """
+    Get the case count for all locations in `location_ids`
+    :returns A dict with location_id as the key and the case count as the value
+    """
+    query = (
         CaseES()
         .domain(domain)
         .owner(location_ids)
-        .user(other_commcare_user_ids)
-        .aggregation(TermsAggregation('by_location', 'owner_id')
-        .size(0))
+        .aggregation(TermsAggregation('by_location', 'owner_id').size(0))
     ).run()
-    return list(set(location_ids) - set(case_query.aggregations.by_location.keys))
-
-
-def _get_location_case_counts_with_single_user(domain, location_ids):
-    query = (CaseES()
-            .domain(domain)
-            .owner(location_ids)
-            .aggregation(TermsAggregation('by_location', 'owner_id')
-            .size(0))
-    ).run()
-    locations = SQLLocation.objects.filter(location_id__in=location_ids)
     counts = query.aggregations.by_location.counts_by_bucket()
-    return locations, counts
+    for location_id in location_ids:
+        if location_id not in counts:
+            counts[location_id] = 0
+
+    return counts
 
 
-def get_locations_with_single_user(domain, location_ids, user_id):
+def get_locations_with_orphaned_cases(domain, location_ids, user_id):
     """
     Takes a list of location IDs and returns all the given location IDs
-    where the user ID is the only one that has cases there.
+    where the user ID is the only one that has cases there. Also searched through descendant locations
+    if a location has `view_descendants` enabled.
```

Comment:
This function returns the case counts for all locations and applicable descendants, not just those that have been orphaned

Reply:
Do you mean it returns case counts for locations that only have one assigned user but no cases as well?

Reply:
Yes, that's right - I don't remember if that's what I meant when I wrote this, but looking again now, you're right - it shows all orphaned locations, whether or not they have cases.  I see you confirmed that this is the desired behavior in another comment:

> Locations with no cases will still be listed in the alert however, as per the ticket's requirements and a discussion I had with Mary. I think the reason for this is to ensure the user is aware of locations that could possibly be orphaned in the future should they ever get cases.

I'm not sold on the utility of that (How would those cases get locations if there aren't users there?  Isn't the same true of all locations that haven't been assigned users?), but if that's the intended behavior, I can confirm that this does look like it behaves correctly.